[BACKLOG-5542] -- updated PromptPanel to not require a parameter defi…

krivera-pentaho opened the pull request:
…nition to be created. Updated API for these changes. Updated unit tests accordingly

krivera-pentaho commented:
@diogofscmariano @annapopova1 @plagoa @pamval @tmorgner please review. 

@tmorgner I made some api changes because prompt panels do not need a parameter definition. To create a new API, an HTML id is required. Also, the operation.render method does not require that HTML id. The operation API creates a prompt panel when it is required and created by the parent API object.

annapopova1 commented on package-res/resources/web/prompting/PromptPanel.js:
```diff
@@ -361,6 +360,19 @@ define(['cdf/lib/Base', 'cdf/Logger', 'dojo/number', 'dojo/i18n', 'common-ui/uti
 
+        getParamDefn: function() {
+          if (!this.paramDefn) {
+            throw 'paramDefn is required. Call setParameterDefinition';
```
1. 'setParameterDefinition' - is not exist. Do you mean PromptPanel#setParamDefn?
2. Need add js doc to 'getParamDefn' and 'setParamDefn' functions

annapopova1 commented on package-res/resources/web/prompting/api/OperationAPI.js:
```diff
@@ -54,18 +54,18 @@ define(['common-ui/prompting/PromptPanel', 'common-ui/prompting/parameters/Param
      *  });
      */
-    this.render = function(id, getParameterXml) {
+    this.render = function(getParameterXml) {
```
Need remove unused parameter in js doc: "@param {String}    id HTML object id where the render the panel"

annapopova1 commented:
Also need change prompting\api\promptApiSample.html page because after this changes it will not work.

krivera-pentaho commented:
@tmorgner I agree and remove the catch.
@annapopova1 I will updatte the promptApiSample, since we are apparently still using it.

krivera-pentaho commented:
@tmorgner I instead checked which exception string is being thrown and if it matches the one i need, then i log a warning, otherwise i re throw the exception.

@diogofscmariano @annapopova1 @plagoa 
I've addressed the issues. Please review.

diogofscmariano commented on package-res/resources/web/prompting/api/OperationAPI.js:
```diff
@@ -26,9 +26,9 @@
+  return function(api, id) {
     this._parameterParser = new ParameterXmlParser();
-    this._promptPanel = null;
+    this._promptPanel = new PromptPanel(id);
```
if we are building a parameter panel without paramDefn will this https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/548/files#diff-fe97c9bc6920bca0d75fa6ad17207ee6R348 always fail? This check should probably not be here or this will never set the value of the auto submit

krivera-pentaho replied:
Yes, it will fail, in the case of the API. Other places can still create the prompt panel with a parameterDefinition as part of the constructor.

diogofscmariano commented on package-res/resources/web/prompting/PromptPanel.js:
```diff
@@ -343,12 +343,11 @@ define(['cdf/lib/Base', 'cdf/Logger', 'dojo/number', 'dojo/i18n', 'common-ui/uti
-          if (!paramDefn) {
-            throw 'paramDefn is required';
-          }
           this.paramDefn = paramDefn;
```
even better, lets remove this from here and use the setParamDefn method! actually it already does what we need

krivera-pentaho replied:
I will call setParamDefn.

diogofscmariano commented:
apart from this comment, everything looks good

krivera-pentaho commented:
@diogofscmariano i have made the changes. Please review.
